pennylane/tape/interfaces/autograd.py

```diff
@@ -107,26 +97,34 @@ def _update_trainable_params(self):
         self.trainable_params = trainable_params
         self._all_parameter_values = params
 
-    def get_parameters(self, trainable_only=True):  # pylint: disable=missing-function-docstring
-        self._update_trainable_params()
-        params = self._all_parameter_values
+    def get_parameters(
+        self, trainable_only=True, return_arraybox=False
+    ):  # pylint: disable=missing-function-docstring
+        params = []
+        iterator = self.trainable_params if trainable_only else self._par_info
 
-        if trainable_only:
-            params = [
-                p
-                for idx, p in enumerate(self._all_parameter_values)
-                if idx in self.trainable_params
-            ]
+        for p_idx in iterator:
+            op = self._par_info[p_idx]["op"]
+            op_idx = self._par_info[p_idx]["p_idx"]
+            params.append(op.data[op_idx])
 
-        return autograd.builtins.list(params)
+return params if return_arraybox else autograd.builtins.list(params)

```

Comment:
nit: Not sure if it's just GitHub screwing up formatting, or is this return statement now outside of the method?

Reply:
that's odd 😆 Just pushed a fix to change it